### PR TITLE
feat(brokers): add uid to can_link_account in FxSyncChannel

### PR DESCRIPTION
**Pull request by eoger**
This connects to https://bugzilla.mozilla.org/show_bug.cgi?id=1383663#c20

### CODE REVIEW

File: app/scripts/models/auth_brokers/fx-sync-channel.js [hunk 1]
```diff
@@ -109,7 +109,9 @@ define(function (require, exports, module) {
 
     beforeSignIn (account) {
       const email = account.get('email');
-      if (this._verifiedCanLinkEmail === email) {
+      const uid = account.get('uid');
+      if (this._verifiedCanLinkEmail === email &&
+          this._verifiedCanLinkUid === uid) {
```
Comment (rfk) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: Sorry for the drive-by, but I wonder whether this should be an `&&` or an `||`.  Do we want to show the "a different user was connected" dialog if they're logging in with a changed email address but the same uid?

Comment (eoger) [reply app/scripts/models/auth_brokers/fx-sync-channel.js]: I think this is ok, we can [make that decision later in the client](http://searchfox.org/mozilla-central/rev/51b3d67a5ec1758bd2fe7d7b6e75ad6b6b5da223/browser/base/content/aboutaccounts/aboutaccounts.js#44-47).
